lib/adagrams.rb

```diff
@@ -0,0 +1,125 @@
+
+require 'csv'
+
+# require_relative '../assets/dictionary-english.csv'
+
+def draw_letters
+letter_pool = %w(
+  A A A A A A A A A
+  B B
+  C C
+  D D D D
+  E E E E E E E E E E E E
+  F F
+  G G G
+  H H
+  I I I I I I I I I
+  J
+  K
+  L L L L
+  M M
+  N N N N N N
+  O O O O O O O O
+  P P
+  Q
+  R R R R R R
+  S S S S
+  T T T T T T
+  U U U U
+  V V
+  W W
+  X
+  Y Y
+  Z
+)
+
+player_hand = letter_pool.sample(10)
+
+return player_hand
+end
+
+
+def uses_available_letters?(input, letters_in_hand)
+  letter_match_counter = 0
+  temp_letters_in_hand = letters_in_hand.dup
+
+  input.chars.each do |letter|
+    if temp_letters_in_hand.include?(letter.upcase)
+      temp_letters_in_hand.delete_at(temp_letters_in_hand.index(letter.upcase))
+      letter_match_counter += 1
+    end
+  end
```

Comment:
Clever way of realizing that counting all "successful matches" with `letter_match_counter` should be the same size as the `input`!

I'll leave this as a "vague thing" to think about... I think that while you iterate through `input.chars`, you can actually determine that `uses_available_letters?` can `return false` before the loop finishes.

Remember, when you use the `return` keyword, it will kick the program execution out of the method early, and return whatever value you give it.